samcli/lib/providers/provider.py

```diff
@@ -371,3 +393,11 @@ def resources(self) -> Dict:
         processed_template_dict: Dict = SamBaseProvider.get_template(self.template_dict, self.parameters)
         resources: Dict = processed_template_dict.get("Resources", {})
         return resources
+
+
+def get_full_path(stack_path: str, logical_id: str) -> str:
+    """
+    Return the unique posix path-like identifier
+    while will used for identify a resource from resources in a multi-stack situation
+    """
+    return posixpath.join(stack_path, logical_id)
```

Comment:
are there any concerns here, where posixpath might behave weirdly trying to look for absolute paths and discard anything else?

Reply:
this `full_path` will not be used in the file system, it is a `posix path-like identifier` used in `sam build` and also in `sam local invoke <full_path>`

Reply:
ok sounds good, the reason I ask is sometimes Pathlib on windows tries to resolve paths before joining them, I wanted to make sure that posixpath does not run into the same issue.

Reply:
Here is the source code of `posixpath.join`, we are pretty safe

https://github.com/python/cpython/blob/fd053fdd39fbdf114b4218ea4309666bafa95788/Lib/posixpath.py#L67-L92